src/test/java/org/fogbowcloud/manager/core/cloudconnector/LocalCloudConnectorTest.java

```diff
@@ -0,0 +1,359 @@
+	    when(this.order.getType()).thenReturn(InstanceType.ATTACHMENT);
+	    when(this.order.getInstanceId()).thenReturn(FAKE_INSTANCE_ID);
+	    when(attachmentPlugin.getInstance(any(String.class), any(Token.class))).thenReturn(this.attachmentInstance);
+		assertEquals(FAKE_INSTANCE_ID, this.localCloudConnector.getInstance(order).getId());
+	}
+	
+	// The order has an InstanceID, so the method getResourceInstance() is called.
+	// addReverseTunnelInfo() is called in this case.
+	@Test
+	public void testgetComputeInstance() throws FogbowManagerException, UnexpectedException {
+		
+		LocalCloudConnector localCloudConnectorSpy = Mockito.spy(this.localCloudConnector);
+		Mockito.doNothing().when(localCloudConnectorSpy).addReverseTunnelInfo(any(String.class), any(ComputeInstance.class));
+
+		this.order = Mockito.mock(ComputeOrder.class);
+	    when(this.order.getType()).thenReturn(InstanceType.COMPUTE);
+	    when(this.order.getInstanceId()).thenReturn(FAKE_INSTANCE_ID);
+	    when(computePlugin.getInstance(any(String.class), any(Token.class))).thenReturn(this.computeInstance);
+		assertEquals(FAKE_INSTANCE_ID, localCloudConnectorSpy.getInstance(order).getId());
+	}
+	
+	
+	// The order doesn't have an InstanceID, so the method an empty NetworkInstance is returned with the same id of order.
+	@Test
+	public void testgetEmptyNetworkInstance() throws FogbowManagerException, UnexpectedException {
+		this.order = Mockito.mock(NetworkOrder.class);
+		when(this.order.getType()).thenReturn(InstanceType.NETWORK);
+		when(this.order.getInstanceId()).thenReturn(null);
+		when(this.order.getId()).thenReturn(FAKE_ORDER_ID);
+		assertEquals(FAKE_ORDER_ID, this.localCloudConnector.getInstance(order).getId());
+	}
+	
+	// The order doesn't have an InstanceID, so the method an empty VolumeInstance is returned with the same id of order.
+	@Test
+	public void testgetEmptyVolumeInstance() throws FogbowManagerException, UnexpectedException {
+		this.order = Mockito.mock(VolumeOrder.class);
+	    when(this.order.getType()).thenReturn(InstanceType.VOLUME);
+	    when(this.order.getInstanceId()).thenReturn(null);
+	    when(this.order.getId()).thenReturn(FAKE_ORDER_ID);
+		assertEquals(FAKE_ORDER_ID, this.localCloudConnector.getInstance(order).getId());
+	}
+	
+	// The order doesn't have an InstanceID, so the method an empty AttachmentInstance is returned with the same id of order.
+	@Test
+	public void testgetEmptyAttachmentInstance() throws FogbowManagerException, UnexpectedException {
+		this.order = Mockito.mock(AttachmentOrder.class);
+	    when(this.order.getType()).thenReturn(InstanceType.ATTACHMENT);
+	    when(this.order.getInstanceId()).thenReturn(null);
+	    when(this.order.getId()).thenReturn(FAKE_ORDER_ID);
+		assertEquals(FAKE_ORDER_ID, this.localCloudConnector.getInstance(order).getId());
+	}
+	
+	// The order doesn't have an InstanceID, so the method an empty ComputeInstance is returned with the same id of order.
```

Comment:
it is very pleasant to have a small description (line 253) of test behaviour. sometimes, mocked stuff is hard to read.

Reply:
I didn't get it. There is a problem in description? 
I've fixed the description by eliminating "the method": "The order doesn't have an InstanceID, so an empty ComputeInstance is returned with the same id of order."

Reply:
no @jvmafra I said it nice to add such description. keep adding them